solr/core/src/java/org/apache/solr/core/SolrConfig.java

```diff
@@ -178,8 +179,12 @@ private class ResourceProvider implements Function<String, InputStream> {
         ZkSolrResourceLoader.ZkByteArrayInputStream zkin =
             (ZkSolrResourceLoader.ZkByteArrayInputStream) in;
         zkVersion = zkin.getStat().getVersion();
-        hash = Objects.hash(zkin.getStat().getCtime(), zkVersion, overlay.getZnodeVersion());
+        hash = Objects.hash(zkin.getStat().getCtime(), zkVersion, overlay.getVersion());
```

Comment:
why is `zkin.getStat().getCtime()` used to compute hash? isn't `zkVersion` not enough?

Reply:
No idea, but that was the behavior before the PR. I'm unsure whether the hash computed here is even used.

Reply:
Note that a re-created file (removed then added) will have its zkVersion reset to 0.  So incorporating CTime guards against this.